File: domain/src/main/java/no/nav/melosys/domain/TidligereMedlemsperiodeId.java
package no.nav.melosys.domain;

import java.io.Serializable;
import java.util.Objects;
import javax.persistence.Column;
import javax.persistence.Embeddable;

@Embeddable
public class TidligereMedlemsperiodeId implements Serializable {

    @Column(name = "behandling_id")
    private Long behandlingId;

    @Column(name = "periode_id")
    private Long periodeId;

    public TidligereMedlemsperiodeId() {}

    public TidligereMedlemsperiodeId(Long behandlingId, Long periodeId) {
        this.behandlingId = behandlingId;
        this.periodeId = periodeId;
    }

    public Long getBehandlingId() {
        return behandlingId;
    }

    public void setBehandlingId(Long behandlingId) {
        this.behandlingId = behandlingId;
    }

    public Long getPeriodeId() {
        return periodeId;
    }

    public void setPeriodeId(Long periodeId) {
        this.periodeId = periodeId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof TidligereMedlemsperiodeId)) {
            return false;
        }
        TidligereMedlemsperiodeId that = (TidligereMedlemsperiodeId) o;
        return Objects.equals(this.behandlingId, that.behandlingId)
            && Objects.equals(this.periodeId, that.periodeId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(behandlingId, periodeId);
    }

}

File: domain/src/main/kotlin/no/nav/melosys/domain/TidligereMedlemsperiodeId.kt
package no.nav.melosys.domain

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import java.io.Serializable
import java.util.*

@Embeddable
class TidligereMedlemsperiodeId(
    @Column(name = "behandling_id")
    var behandlingId: Long? = null,

    @Column(name = "periode_id")
    var periodeId: Long? = null
) : Serializable {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is TidligereMedlemsperiodeId) return false

        return Objects.equals(behandlingId, other.behandlingId) &&
            Objects.equals(periodeId, other.periodeId)
    }

    override fun hashCode(): Int = Objects.hash(behandlingId, periodeId)
}
